source/fundamentals/data-formats/document-data-format-pojo.txt

```diff
@@ -13,17 +13,215 @@ Document Data Format: POJOs
+     field.
+
+Configure the Driver for POJOs
+------------------------------
+
+To set up the driver to store and retrieve POJOs, we need to specify:
+
+- A codec provider that contains :doc:`Codecs </fundamentals/data-formats/codecs>`
+  that define how to encode/decode the data between the POJO and MongoDB
+  document, and which POJO classes or packages that the Codecs should apply to.
+- A :java-docs:`CodecRegistry <apidocs/bson/org/bson/codecs/configuration/CodecRegistry.html>`
+  instance that contains the aforementioned information.
+- A ``MongoClient``, ``MongoDatabase``, or ``MongoCollection`` instance
+  configured to use the ``CodecRegistry``.
+- A ``MongoCollection`` instance configured to cast to your POJO document
+  class.
+
+Follow the steps below to see how to perform each of the requirements:
+
+
+1. Configure the :java-docs:`PojoCodecProvider </apidocs/bson/org/bson/codecs/pojo/PojoCodecProvider.html>`,
+   a codec provider that provides the Codecs that define how to encode the data
+   and on which classes to apply the Codecs. In this example, we use the
+   :java-docs:`automatic(true) </apidocs/bson/org/bson/codecs/pojo/PojoCodecProvider.Builder.html#automatic(boolean)>`
+   setting of the :java-docs:`PojoCodecProvider.Builder </apidocs/bson/org/bson/codecs/pojo/PojoCodecProvider.Builder.html>`
+   to apply the Codecs to any class and its properties.
+
+   .. code-block:: java
+
+      CodecProvider pojoCodecProvider = PojoCodecProvider.builder().automatic(true).build();
+
+.. note::
+
+   Codec providers also contain other objects such as ``ClassModel`` and
+   ``Convention`` instances that further define the serialization behavior.
+   For more information on codec providers and customization, see our guide
+   on :doc:`POJO Customization </fundamentals/data-formats/pojo-customization>`.
+
+
+2. Add the ``PojoCodecProvider`` instance to a ``CodecRegistry``. The
+   ``CodecRegistry`` allows you to specify one or more Codec providers to
+   encode the POJO data. See the code below to see how to instantiate the
+   ``CodecRegistry`` instance:
+
+   .. code-block:: java
+
+      // ensure you use the following static imports above your class definition
+      import static com.mongodb.MongoClientSettings.getDefaultCodecRegistry;
+      import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+      import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+      // ...
+
+      CodecRegistry pojoCodecRegistry = fromRegistries(getDefaultCodecRegistry(), fromProviders(pojoCodecProvider));
+
+3. Configure our ``MongoClient``, ``MongoDatabase``, or ``MongoCollection``
+   instance to use the Codecs in the ``CodecRegistry``. You only need to
+   configure one of these. In this example, we set it on a ``MongoDatabase``
+   called ``sample_pojos`` using the ``withCodecRegistry()`` method.
+
+   .. code-block:: java
+
+      try (MongoClient mongoClient = MongoClients.create(uri)) {
```

Comment:
Considering how expensive each `MongoClient` is, would we be better off leaving the try-with-resources pattern out, since most end users probably won't use a `MongoClient` like that?

Reply:
Thanks for noting this. I think Jeff decided he preferred this pattern for sample code in our docs (except for when demonstrating MongoClientSettings.builder()), and it's been the pattern throughout our documentation. I think it could be useful to demonstrate efficient code if we had a tutorial on building production apps or making your code more efficient.